sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -269,14 +269,17 @@ def _allocate_parameter_mem(
                 self._standard_intercept.shape, dtype=np.float64, order="C"
             )
 
-    def _make_validation_split(self, y):
+    def _make_validation_split(self, y, sample_weight):
```

Comment:
I think the code would be easier to follow if this variable was called `sample_mask` and that this method would be called as:

```python
self._make_validation_split(y, sample_mask=sample_weight > 0)
```

otherwise the caller might expect that the sampling of the validation set would have probabilities proportional to the weights which is not the case in our shuffle-based CV splitters.

Reply:
Updated as suggested :)